# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1347921269 14400
#      Mon Sep 17 18:34:29 2012 -0400
# Node ID 231d8d3b82257852f001b00d0a224b4680181b6a
# Parent  1054ab58cd58531c3f00eb089d32e4ab6a145771
provide command_hist::clear_history function

* cmd-hist.h, cmd-hist.cc (command_history::clear,
command_history::do_clear): New functions.
* cmd-hist.cc (gnu_history::do_clear): New function.
* oct-rl-hist.h, oct-rl-hist.c (octave_clear_history): New function.

diff --git a/liboctave/util/cmd-hist.cc b/liboctave/util/cmd-hist.cc
--- a/liboctave/util/cmd-hist.cc
+++ b/liboctave/util/cmd-hist.cc
@@ -64,16 +64,18 @@ public:
   void do_process_histcontrol (const std::string&);
 
   std::string do_histcontrol (void) const;
 
   void do_add (const std::string&);
 
   void do_remove (int);
 
+  void do_clear (void);
+
   int do_where (void) const;
 
   int do_length (void) const;
 
   int do_max_input_history (void) const;
 
   int do_base (void) const;
 
@@ -196,16 +198,22 @@ gnu_history::do_add (const std::string& 
 }
 
 void
 gnu_history::do_remove (int n)
 {
   ::octave_remove_history (n);
 }
 
+void
+gnu_history::do_clear (void)
+{
+  ::octave_clear_history ();
+}
+
 int
 gnu_history::do_where (void) const
 {
   return ::octave_where_history ();
 }
 
 int
 gnu_history::do_length (void) const
@@ -582,16 +590,23 @@ command_history::add (const std::string&
 
 void
 command_history::remove (int n)
 {
   if (instance_ok ())
     instance->do_remove (n);
 }
 
+void
+command_history::clear (void)
+{
+  if (instance_ok ())
+    instance->do_clear ();
+}
+
 int
 command_history::where (void)
 {
   return (instance_ok ())
     ? instance->do_where () : 0;
 }
 
 int
@@ -802,16 +817,21 @@ command_history::do_add (const std::stri
 {
 }
 
 void
 command_history::do_remove (int)
 {
 }
 
+void
+command_history::do_clear (void)
+{
+}
+
 int
 command_history::do_where (void) const
 {
   return 0;
 }
 
 int
 command_history::do_length (void) const
diff --git a/liboctave/util/cmd-hist.h b/liboctave/util/cmd-hist.h
--- a/liboctave/util/cmd-hist.h
+++ b/liboctave/util/cmd-hist.h
@@ -60,16 +60,18 @@ public:
   static void ignore_entries (bool = true);
 
   static bool ignoring_entries (void);
 
   static void add (const std::string&);
 
   static void remove (int);
 
+  static void clear (void);
+
   static int where (void);
 
   static int length (void);
 
   static int max_input_history (void);
 
   static int base (void);
 
@@ -153,16 +155,18 @@ protected:
   virtual void do_ignore_entries (bool);
 
   virtual bool do_ignoring_entries (void) const;
 
   virtual void do_add (const std::string&);
 
   virtual void do_remove (int);
 
+  virtual void do_clear (void);
+
   virtual int do_where (void) const;
 
   virtual int do_length (void) const;
 
   virtual int do_max_input_history (void) const;
 
   virtual int do_base (void) const;
 
diff --git a/liboctave/util/oct-rl-hist.c b/liboctave/util/oct-rl-hist.c
--- a/liboctave/util/oct-rl-hist.c
+++ b/liboctave/util/oct-rl-hist.c
@@ -205,16 +205,22 @@ octave_remove_history (int n)
     {
       if (discard->line)
         free (discard->line);
 
       free (discard);
     }
 }
 
+void
+octave_clear_history (void)
+{
+  clear_history ();
+}
+
 char *
 octave_history_goto_mark (int n)
 {
   HIST_ENTRY *h;
 
   char *retval = 0;
 
   if (history_set_pos (n))
diff --git a/liboctave/util/oct-rl-hist.h b/liboctave/util/oct-rl-hist.h
--- a/liboctave/util/oct-rl-hist.h
+++ b/liboctave/util/oct-rl-hist.h
@@ -61,16 +61,18 @@ extern int octave_read_history_range (co
 extern int octave_write_history (const char *);
 
 extern int octave_append_history (int, const char *);
 
 extern int octave_history_truncate_file (const char *, int);
 
 extern void octave_remove_history (int);
 
+extern void octave_clear_history (void);
+
 extern char *octave_history_goto_mark (int n);
 
 extern char *octave_history_get (int n);
 
 extern char **octave_history_list (int, int);
 
 extern void octave_replace_history_entry (int, const char *);
 
